srml/balances/src/lib.rs

```diff
@@ -1171,19 +1182,23 @@ impl<T: Trait<I>, I: Instance> TakeFees<T, I> {
 	///   - (optional) _tip_: if included in the transaction, it will be added on top. Only signed
 	///      transactions can have a tip.
 	fn compute_fee(len: usize, info: DispatchInfo, tip: T::Balance) -> T::Balance {
-		// length fee
 		let len_fee = if info.pay_length_fee() {
 			let len = T::Balance::from(len as u32);
 			let base = T::TransactionBaseFee::get();
-			let byte = T::TransactionByteFee::get();
-			base.saturating_add(byte.saturating_mul(len))
+			let per_byte = T::TransactionByteFee::get();
+			base.saturating_add(per_byte.saturating_mul(len))
 		} else {
 			Zero::zero()
 		};
 
-		// weight fee
-		let weight = info.weight;
-		let weight_fee: T::Balance = <system::Module<T>>::next_weight_multiplier().apply_to(weight).into();
+		let weight_fee = {
+			// cap the weight to the maximum defined in runtime, otherwise it will be the `Bounded`
+			// maximum of its data type, which is not desired.
+			let capped_weight = info.weight.min(<T as system::Trait>::MaximumBlockWeight::get());
```

Comment:
this kinda sucks that every signed extension using weight must cap it itself if they want to. This is because in the details of the `WeighData` we don't know the `MaximumBlockWeight`. Might be able to improve it by further abstracting `type Weight: u32` into some other custom type and give it a custom implementation of `Bounded` by the runtime.

Reply:
I also do the same in system `CheckWeight`. Might be debatable. 

I had the assumption that the weight fo one individual tx cannot go beyond the weight of the entire block. Even if thee annotationa sets the weight to infinity the runtime checks interpret it as whatever `MaximumBlockWeight` of the node is. 

If this does not exist, then `Max` variants of weight are actually a footgun, using it will cause a much higher fee and won't bring any benefit.